src/components/tables/TableThunkActionCreators.ts

```diff
@@ -0,0 +1,151 @@
+        let shouldKeep = false;
+
+        tableOwnProps.headingAttributes.forEach((headingAttribute: ITableHeadingAttribute) => {
+          const { attributeName, attributeFormatter } = headingAttribute;
+          const attributeValue = tableDataById[dataId][attributeName];
+          const attributeValueToUse = attributeFormatter ? attributeFormatter(attributeValue) : attributeValue;
+          shouldKeep =
+            shouldKeep
+            || attributeValueToUse
+              .toString()
+              .toLowerCase()
+              .indexOf(tableState.filter.toLowerCase()) > -1;
+        });
+
+        return shouldKeep;
+      };
+
+      const filterMethod = tableOwnProps.filterMethod
+        ? (dataId: string): boolean => tableOwnProps.filterMethod(tableDataById[dataId], tableOwnProps)
+        : filterDefault;
+
+      nextDisplayedIds = nextDisplayedIds.filter(filterMethod);
+    }
+
+    totalEntries = nextDisplayedIds.length;
+    totalPages = Math.ceil(totalEntries / tableState.perPage);
+
+    // pagination logic
+    const startingIndex = tableState.page * tableState.perPage;
+    const endingIndex = startingIndex + tableState.perPage;
+    nextDisplayedIds = nextDisplayedIds.slice(startingIndex, endingIndex);
+
+    // sort default logic
+    const { sortState } = tableState;
+    if (sortState && sortState.order !== TableSortingOrder.UNSORTED) {
+      const defaultSortBy = (displayedId: string) => {
+        const cleanAttributeValue = convertUndefinedAndNullToEmptyString(tableDataById[displayedId][sortState.attribute]);
+        return cleanAttributeValue.toString().toLowerCase();
+      };
+
+      const sortByMethod = _.findWhere(tableOwnProps.headingAttributes, { attributeName: sortState.attribute }).sortByMethod || defaultSortBy;
+
+      nextDisplayedIds = _.sortBy(nextDisplayedIds, sortByMethod);
+
+      if (sortState.order === TableSortingOrder.DESCENDING) {
+        nextDisplayedIds.reverse();
+      }
+    }
+
+    return {
+      ...tableState,
+      data: {
+        ...tableState.data,
+        displayedIds: nextDisplayedIds,
+        totalEntries,
+        totalPages,
+      },
+      page: shouldResetPage ? 0 : tableState.page,
+    };
+  };
+};
+
+export const defaultTableStateModifyerThunk = (tableOwnProps: ITableOwnProps, shouldResetPage: boolean) => {
+  return (dispatch: any) => {
+    dispatchPreTableStateModification(tableOwnProps, dispatch);
+    dispatch(modifyState(tableOwnProps.id, defaultTableStateModifyer(tableOwnProps, shouldResetPage)));
+    dispatchPostTableStateModification(tableOwnProps, dispatch);
+  };
+};
+
+export const serverTableStateModifyerThunk = (tableOwnProps: ITableOwnProps, shouldResetPage: boolean) => {
+  return (dispatch: any, getState: () => { [globalStateProp: string]: any; tables: ITablesState; }) => {
+    dispatchPreTableStateModification(tableOwnProps, dispatch);
+    $.get(tableOwnProps.serverMode.url(getState().tables[tableOwnProps.id], tableOwnProps))
```

Comment:
I'm not quite sure this is the kind of thing that should be in React Vapor

Reply:
@GermainBergeron had doubts too, we could mirror https://react-table.js.org/#/story/server-side-data and just add an `onFetchData` that would be triggered where` this.props.onModifyData` is currently triggered. A bit like @GermainBergeron suggested: if you pass a `onFetchData` prop (which i think would be a dispatchable thunk action creator), the this.props.onModifyData won't be triggered and onFetchData will be dispatched instead, and thus the user of the component will have to trigger the state modification inside onFetchData.  If we implement this, I think customMode and serverMode could be removed and onFetchData will replace both.

Reply:
it could be more generic, like `onManualModifyData`